python/test/unit/test_diffusion.py

```diff
@@ -0,0 +1,315 @@
+                tag=0,
+            )
+
+            labels = A.label_dict(
+                {
+                    "soma-region": "(tag 0)",
+                    "soma-center": '(on-components 0.5 (region "soma-region"))',
+                    "soma-end": '(on-components 1.0 (region "soma-region"))',
+                }
+            )
+        elif num_segs == 2:
+            s = tree.append(
+                A.mnpos,
+                A.mpoint(-length / 2, 0, 0, radius_1),
+                A.mpoint(0, 0, 0, radius_1),
+                tag=0,
+            )
+            _ = tree.append(
+                s,
+                A.mpoint(0, 0, 0, radius_2),
+                A.mpoint(+length / 2, 0, 0, radius_2),
+                tag=1,
+            )
+
+            labels = A.label_dict(
+                {
+                    "soma-region": "(tag 0)",
+                    "dendriteA-region": "(tag 1)",
+                    "soma-center": '(on-components 0.5 (region "soma-region"))',
+                    "soma-end": '(on-components 1.0 (region "soma-region"))',
+                    "dendriteA-center": '(on-components 0.5 (region "dendriteA-region"))',
+                }
+            )
+        elif num_segs == 3:
+            s = tree.append(
+                A.mnpos,
+                A.mpoint(-1 / 3 * length, 0, 0, radius_1),
+                A.mpoint(0, 0, 0, radius_1),
+                tag=0,
+            )
+            _ = tree.append(
+                s,
+                A.mpoint(0, 0, 0, radius_2),
+                A.mpoint(+1 / 3 * length, 0, 0, radius_2),
+                tag=1,
+            )
+            _ = tree.append(
+                s,
+                A.mpoint(-1 / 3 * length, 0, 0, radius_3),
+                A.mpoint(-2 / 3 * length, 0, 0, radius_3),
+                tag=2,
+            )
+
+            labels = A.label_dict(
+                {
+                    "soma-region": "(tag 0)",
+                    "dendriteA-region": "(tag 1)",
+                    "dendriteB-region": "(tag 2)",
+                    "soma-center": '(on-components 0.5 (region "soma-region"))',
+                    "soma-end": '(on-components 1.0 (region "soma-region"))',
+                    "dendriteA-center": '(on-components 0.5 (region "dendriteA-region"))',
+                    "dendriteB-center": '(on-components 0.5 (region "dendriteB-region"))',
+                }
+            )
+        else:
+            raise ValueError(
+                f"Specified number of segments ({num_segs}) not supported."
+            )
+        morph = A.morphology(tree)
+
+        # ---------------------------------------------------------------------------------------
+        # decorate the morphology with mechanisms
+        dec = A.decor()
+        if num_segs < 3:
+            dec.discretization(
+                A.cv_policy(f"(fixed-per-branch {num_segs*num_cvs_per_seg} (branch 0))")
+            )
+        elif num_segs == 3:
```

Comment:
Does this result in different numbers of CVs? It seems like the proposed
morphology produces either one linear or  three branches in Y-shape. In both cases you're setting `num_cvs_per_seg` on all branches. Thus, this should be equivalent to just setting `fixed-per-branch`?!

Reply:
Thanks for pointing that out. Indeed the branch specification is unnecessary here, I've removed it from the code. But the distinction between `num_segs < 3` and `num_segs == 3` still is necessary because for less than three segments, there is only one branch.

Reply:
True. Question though: Do you want a fixed number/branch or really a fixed $\Delta x$?

Reply:
Yeah, I wanted to control the number of CVs, however, I guess here it shouldn't matter which quantity is kept fixed.

Reply:
That seemed a bit unusual and so I am asking.